### PR TITLE
RMMIS-6910: Allow Required Indicator to be on the left.

### PR DESCRIPTION
**Pull request by traviscooperscdhhs**

### CODE REVIEW

**Comment by kjschulz** [comment]
sadly and a bit begrudgingly, :+1:

File: src/styles/Field.less [hunk 1]
```diff
@@ -34,6 +29,13 @@
   }
 }
 
+.required {
+  margin-left: @padding-xs-horizontal;
```
Comment (mallenscdhhs) [inline src/styles/Field.less]: if required indicator is on the left, then we don't want a margin on the left

File: src/Terms.jsx [hunk 2]
```diff
@@ -85,6 +85,7 @@ class Terms extends React.Component {
           disabled={!this.props.termsRead}
           label={this.props.checkboxLabel}
           required={this.props.required}
+          requiredIndicatorPosition={this.props.requiredIndicatorPosition}
```
Comment (mallenscdhhs) [inline src/Terms.jsx]: shouldn't there be a `defaultProp` for `requiredIndicatorPosition` for this component?

Comment (traviscooperscdhhs) [inline src/Terms.jsx]: for Terms or for FieldLabel

Comment (traviscooperscdhhs) [inline src/Terms.jsx]: Added for both.

File: src/styles/Field.less (hunk 1 above)
Comment (traviscooperscdhhs) [inline src/styles/Field.less]: updated.

File: src/Terms.jsx [hunk 3]
```diff
@@ -111,7 +113,8 @@ Terms.defaultProps = {
   checkboxLabel: '',
   className: '',
   required: true,
-  termsRead: false
+  termsRead: false,
+  requiredIndicatorPosition: ''
```
Comment (mallenscdhhs) [inline src/Terms.jsx]: Should this be "left", or "right"?

Comment (traviscooperscdhhs) [inline src/Terms.jsx]: empty is fine. it only checks for Left to pull something left. so Undefined, null, '' would work for 'right'.

**Pull request merged**